test/services/schema-object-factory.spec.ts

```diff
@@ -412,6 +412,24 @@ describe('SchemaObjectFactory', () => {
       expect(Object.keys(schemas)).toContain('UpdateUserDto');
     });
 
+    it('should override the schema name of base class', () => {
+      @ApiSchema({
+        name: 'CreateUser'
+      })
+      class CreateUserDto {}
+
+      @ApiSchema({
+        name: 'UpdateUser'
+      })
```

Comment:
Interesting enough, the case _without_ an `@ApiSchema` decorator in the child class has been tested. Just not the case when both the parent and the child class have a decorator set.